lightwood/api/types.py

```diff
@@ -542,3 +527,54 @@ class ModelAnalysis:
     confusion_matrix: object
     histograms: object
     dtypes: object
+
+
+@dataclass
+class PredictionArguments:
```

Comment:
Can we remove `predict_proba` and instead add it as an argument here? It's fine if the `predict` interface changes to some extent we can always have the return type be a `Union` of the various types returned by the various modes.

Reply:
Also can you add a "all_mixers" mode or something like that which returns a prediction for every mixer? We needed it for the benchmarks anyway

Reply:
Both are really good ideas, let me take a stab at them.